src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala

```diff
@@ -22,7 +22,10 @@ object AgoraMongoDao {
   val KeySeparator = ":"
   val DefaultFindProjection = Option(new AgoraEntityProjection(Seq.empty[String], Seq[String]("payload", "documentation")))
 
-  def EntityToMongoDbObject(entity: AgoraEntity): DBObject = JSON.parse(entity.toJson.toString()).asInstanceOf[DBObject]
+  def EntityToMongoDbObject(entity: AgoraEntity): DBObject = {
+    //we need to remove url if it exists since we don't store it in the document
```

Comment:
I think the reasoning is that it has configurable components. (port and hostname for instance) Because of that we would potentially be storing urls that become invalid and we have no way of changing them for older entities since we are write once.